src/js/helpers/transaction-queue.js

```diff
@@ -0,0 +1,54 @@
+import { state, updateState } from './state'
+
+export function addTransactionToQueue(txObject) {
+  const { transactionQueue } = state
+  const newQueueState = [...transactionQueue, txObject]
+  updateState({ transactionQueue: newQueueState })
+
+  return newQueueState
+}
+
+export function removeTransactionFromQueue(id) {
+  const { transactionQueue } = state
+  const newQueueState = transactionQueue.filter(
+    txObj => txObj.transaction.id !== id
+  )
+  updateState({ transactionQueue: newQueueState })
+
+  return newQueueState
+}
+
+export function updateTransactionInQueue(id, update) {
+  const txObj = getTxObjFromQueue(id)
+  txObj.transaction = Object.assign(txObj.transaction, update)
+
+  return txObj
+}
+
+export function getTxObjFromQueue(id) {
+  const { transactionQueue } = state
+  return transactionQueue.find(txObj => txObj.transaction.id === id)
+}
+
+export function isDuplicateTransaction({ value, to }) {
+  const { transactionQueue } = state
+
+  return Boolean(
+    transactionQueue.find(
+      txObj => txObj.transaction.value === value && txObj.transaction.to === to
```

Comment:
Not sure if I understand what's happening here 100%, but are these checks ok? What if a user wanted to make calls to different methods on a single contract? The `to` and `value` properties (I think?) would be the same across, but the txs would not be duplicates

Reply:
Yeah that's a good point, I will add a check to see if it is also a contract send and if it is, check to see if it is the same method that is being called.

Reply:
@liamaharon you're correct - a check across the `data` property of the objects would make sense here

Reply:
Changes added in commits:
- 756826c 
- d470a52